File: src/AAS.Domain/Bids/Bid.cs
using AAS.Domain.Bids.Enums;
using AAS.Tools.Types.IDs;
using File = AAS.Domain.Files.File;

namespace AAS.Domain.Bids;

public class Bid
{
    public ID Id { get; }
    public int Number { get; }
    public string Title { get; }
    public string? Description { get; }
    public string? DenyDescription { get; }
    public File[] Files { get; }
    public BidStatus Status { get; }
    public DateOnly? AcceptanceDate { get; }
    public DateOnly? ApproximateDate { get; }
    public ID CreatedUserId { get; }

    public Bid(ID id, int number, string title, string? description, string? denyDescription, File[] files, BidStatus status,
        DateOnly? acceptanceDate, DateOnly? approximateDate, ID createdUserId)
    {
        Id = id;
        Number = number;
        Title = title;
        Description = description;
        DenyDescription = denyDescription;
        Files = files;
        Status = status;
        AcceptanceDate = acceptanceDate;
        ApproximateDate = approximateDate;
        CreatedUserId = createdUserId;
    }
}

File: src/AAS.BackOffice/client_app/src/domain/bids/bid.ts
import { File, mapToFiles } from "../files/file";

export class Bid {
    constructor(
        public readonly id: string,
        public readonly number: number,
        public readonly title: string,
        public readonly description: string,
        public readonly denyDescription: string | null,
        public readonly files: File[],
        public readonly status: number,
        public readonly acceptanceDate: Date | null,
        public readonly approximateDate: Date | null,
        public readonly createdUserId: string
    ) {
    }
}

export function toBid(value: any): Bid {
    return new Bid(
        value.id,
        value.number,
        value.title,
        value.description,
        value.denyDescription,
        mapToFiles(value.files),
        value.status,
        value.acceptanceDate,
        value.approximateDate,
        value.createdUserId
    )
}

export function toBids(values: any[]): Bid[] {
    return values.map(toBid);
}
